pkg/trace/stats/aggregation.go

```diff
@@ -77,16 +78,23 @@ func clientOrProducer(spanKind string) bool {
 // NewAggregationFromSpan creates a new aggregation from the provided span and env
 func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregationKey, enablePeerTagsAgg bool, peerTagKeys []string) (Aggregation, []string) {
 	synthetics := strings.HasPrefix(origin, tagSynthetics)
+	var isTraceRoot pb.TraceRootFlag
+	if s.ParentID == 0 {
+		isTraceRoot = pb.TraceRootFlag_TRUE
+	} else {
+		isTraceRoot = pb.TraceRootFlag_FALSE
+	}
 	agg := Aggregation{
 		PayloadAggregationKey: aggKey,
 		BucketsAggregationKey: BucketsAggregationKey{
-			Resource:   s.Resource,
-			Service:    s.Service,
-			Name:       s.Name,
-			SpanKind:   s.Meta[tagSpanKind],
-			Type:       s.Type,
-			StatusCode: getStatusCode(s),
-			Synthetics: synthetics,
+			Resource:    s.Resource,
+			Service:     s.Service,
+			Name:        s.Name,
+			SpanKind:    s.Meta[tagSpanKind],
+			Type:        s.Type,
+			StatusCode:  getStatusCode(s),
+			Synthetics:  synthetics,
+			IsTraceRoot: isTraceRoot,
```

Comment:
```suggestion
			IsTraceRoot: s.ParentID == 0,
```

Might be cleaner that way. You can then remove the variable up above too.

Reply:
Because I changed `isTraceRoot` to an enum I don't think I can change it to assign like this

Reply:
Ah good point.